src/main/scala/io/laserdisc/zio/oci/objectstorage/Live.scala

```diff
@@ -0,0 +1,67 @@
+package io.laserdisc.zio.oci.objectstorage
+
+import com.oracle.bmc.model.BmcException
+import com.oracle.bmc.objectstorage.ObjectStorageAsyncClient
+import com.oracle.bmc.objectstorage.responses.{GetObjectResponse, ListBucketsResponse, ListObjectsResponse}
+import com.oracle.bmc.objectstorage.requests.{GetObjectRequest, ListBucketsRequest, ListObjectsRequest}
+import com.oracle.bmc.responses.AsyncHandler
+import zio._
+import zio.blocking.Blocking
+import zio.duration._
+import zio.stream.{Stream, ZStream}
+
+import java.util.concurrent.{Future => JFuture}
+import java.io.IOException
+
+final class Live(unsafeClient: ObjectStorageAsyncClient) extends ObjectStorage.Service {
+  override def listBuckets(compartmentId: String, namespace: String): IO[BmcException, ListBucketsResponse] =
+    listBuckets(ListBucketsRequest.builder().compartmentId(compartmentId).namespaceName(namespace).build())
+
+  private def listBuckets(request: ListBucketsRequest): IO[BmcException, ListBucketsResponse] =
+    execute[ListBucketsRequest, ListBucketsResponse] { c =>
+      c.listBuckets(request, _: AsyncHandler[ListBucketsRequest, ListBucketsResponse])
+    }
+
+  override def listObjects(namespace: String, bucket: String): IO[BmcException, ListObjectsResponse] =
+    execute[ListObjectsRequest, ListObjectsResponse] { c =>
+      c.listObjects(
+        ListObjectsRequest.builder().bucketName(bucket).namespaceName(namespace).build(),
+        _: AsyncHandler[ListObjectsRequest, ListObjectsResponse]
+      )
+    }
+
+  override def getObject(namespace: String, bucket: String, name: String): ZStream[Blocking, BmcException, Byte] =
+    ZStream
+      .fromEffect(execute[GetObjectRequest, GetObjectResponse] { c =>
+        c.getObject(
+          GetObjectRequest.builder().namespaceName(namespace).bucketName(bucket).objectName(name).build(),
+          _: AsyncHandler[GetObjectRequest, GetObjectResponse]
+        )
+      })
+      .flatMap(is => ZStream.fromInputStreamEffect(IO(is.getInputStream()).refineToOrDie[IOException]))
+      .refineOrDie { case e: BmcException =>
+        e
+      }
```

Comment:
```suggestion
      .refineToOrDie[BmcException]
```
Should be available, see https://github.com/zio/zio/blob/c0f8acd5a15b08f4b9fca87d080b2691a952fd50/streams/shared/src/main/scala/zio/stream/ZStream.scala#L4261